Sources/Controllers/Map/Widgets/OASunriseSunsetWidget.mm

```diff
@@ -369,4 +372,24 @@ - (NSString *) formatNextTime:(NSTimeInterval)nextTime
     return [dateFormatter stringFromDate:nextDate];
 }
 
+- (void) updateCachedLocation
+{
+    OsmAnd::LatLon centerLocation = OsmAnd::Utilities::convert31ToLatLon([OARootViewController instance].mapPanel.mapViewController.mapView.target31);
```

Comment:
It's more optimal to cache target31, this way you won't have to create a new CLLocation object each time